langserver/definition.go

```diff
@@ -40,11 +58,32 @@ func (h *LangHandler) handleDefinition(ctx context.Context, conn JSONRPC2Conn, r
 	if len(nodes) == 0 {
 		return nil, errors.New("definition not found")
 	}
-	locs := goRangesToLSPLocations(fset, nodes)
-	for i := range locs {
+	locs := make([]lspext.LocationInformation, 0, len(nodes))
+	for _, node := range nodes {
+		// Determine location information for the node.
+		l := lspext.LocationInformation{
+			Location: goRangeToLSPLocation(fset, node.Pos(), node.End()),
+		}
 		// LSP expects a range to be of the entire body, not just of the
 		// identifier, so we pretend its just a position and not a range.
-		locs[i].Range.End = locs[i].Range.Start
+		l.Location.Range.End = l.Location.Range.Start
+
+		// Determine metadata information for the node.
+		//rootFile := pkg.Files[0] // TODO: is this correct?
+		def, err := refs.DefInfo(pkg.Pkg, &pkg.Info, pathEnclosingInterval, node.Pos())
```

Comment:
Since `def` is only used within the following if-else block, a clearer way to write this would be:

```
if def, err := refs.DefInfo(pkg.Pkg, &pkg.Info, pathEnclosingInterval, node.Pos()); err == nil {

} else 

}
```